### PR TITLE
Check for valid service override endpoint type on create and update

### PR DESCRIPTION
**Pull request by stuggi**
Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/505

### CODE REVIEW

**Review by gibizer — COMMENTED** [review]
Looks good. Extra thanks for the webhook envtest. Once the dependency lands, and we are out of the freeze I'm OK to land this.

File: api/v1beta1/placementapi_webhook.go [hunk 1]
```diff
@@ -126,19 +127,33 @@ func (r *PlacementAPI) ValidateDelete() (admission.Warnings, error) {
...
+
+	return allErrs
 }
 
 func (r PlacementAPISpecCore) ValidateUpdate(old PlacementAPISpecCore, basePath *field.Path) field.ErrorList {
-	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+	var allErrs field.ErrorList
+
+	// validate the service override key is valid
+	allErrs = append(allErrs, service.ValidateRoutedOverrides(basePath, r.Override.Service)...)
```
Comment (gibizer) [inline api/v1beta1/placementapi_webhook.go]: I'm wondering if ValidateRoutedOverrides could be more generic if the caller pass a path that is already in sync with the field value passed in. E.g. if in a specific service operator the override service field is somewhere else in the instance then the following patter can keep the lib-common code ignorant about that:
```golang
	allErrs = append(allErrs, service.ValidateRoutedOverrides(basePath.Child("override").Child("service"), r.Override.Service)...)
```

Comment (stuggi) [reply api/v1beta1/placementapi_webhook.go]: done

**Review by gibizer — APPROVED** [review]

**Pull request merged**